test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_simultaneous.cpp

```diff
@@ -92,33 +91,39 @@ struct OutOfOrderTest : public BasicMutableCommandBufferTest
         return CL_SUCCESS;
     }
 
-    //--------------------------------------------------------------------------
     cl_int SetUp(int elements) override
     {
         cl_int error = BasicMutableCommandBufferTest::SetUp(elements);
         test_error(error, "BasicMutableCommandBufferTest::SetUp failed");
 
-        error = SetUpKernel();
-        test_error(error, "SetUpKernel failed");
-
-        out_of_order_queue = clCreateCommandQueue(
-            context, device, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, &error);
-        test_error(error, "Unable to create command queue to test with");
-
-        cl_command_buffer_properties_khr properties[3] = {
-            CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_MUTABLE_KHR, 0
-        };
-
-        out_of_order_command_buffer = clCreateCommandBufferKHR(
-            1, &out_of_order_queue, properties, &error);
-        test_error(error, "clCreateCommandBufferKHR failed");
+        if (out_of_order_request)
+        {
+            work_queue = clCreateCommandQueue(
+                context, device, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
+                &error);
+            test_error(error, "Unable to create command queue to test with");
+
+            cl_command_buffer_properties_khr properties[3] = {
+                CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_MUTABLE_KHR, 0
+            };
+
+            work_command_buffer =
+                clCreateCommandBufferKHR(1, &work_queue, properties, &error);
```

Comment:
I think we need to set `CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR` as well for `simultaneous_use_requested`, here and in the in-order queue case